### ISSUE TITLE
Improve outline selection

### ISSUE THREAD

**Issue by erikmannerfelt** [eliskasieglova/SvalbardSurges#3]
You bring up a good point with this comment!:
https://github.com/eliskasieglova/SvalbardSurges/blob/806d6e2405360d3b4c9ab857846a7373125f88e7/main.py#L105

There might be cases of duplicate names. This isn't great, and as you say, requires some kind of handling.

I see two solutions, where the choice between them almost needs a real-world example:
1. Merge the two outlines (perhaps only if they touch?)
2. Pick the largest outline.

If you want to filter by the largest, it's pretty fast in (geo-)pandas:
```python
gao_glacier["area"] = gao_glacier.geometry.area

return gao_glacier.sort_values("area").iloc[-1]
```
This first creates an area column (maybe it already exists? But best to make a new one in case we change outlines). Then it sorts by area, and extracts the last (largest) row. Note that this will fail with a very confusing error if the `gao.query(...)` call was empty. Therefore, after the query, I suggest to add a check that there is at least one outline in the filtered data:
```python
assert gao_glacier.shape[0] > 0, f"Query NAME=='{glacier_name}' returned zero features."
```

If you want to merge all touching outlines, you can run these two:
```python
gao_glacier = gao_glacier.dissolve().explode()
```
This will first take the unary union of all outlines (`dissolve()`) and then separate all outlines that don't touch into separate rows (`explode()`). Note that most metadata are lost using this approach. But we don't need that right now, so I guess it's fine!

**Comment by eliskasieglova** [comment]
Here is an example with Scheelebreen: it has three parts that are not touching each other and picking out only the largest one would make us lose the two smaller parts (or at least one of them).

<img width="241" alt="image" src="https://github.com/eliskasieglova/SvalbardSurges/assets/57137417/8c591aea-d92b-4301-8239-41797fd10c39">


In the code I selected all three polygons, which were then used to mask the DEM and it worked fine:

<img width="144" alt="image" src="https://github.com/eliskasieglova/SvalbardSurges/assets/57137417/0c172787-ee32-40d1-949a-4dbc9cb25486">

So as long as the three glaciers with the same name are actually the same glacier (like in this example), this isn't really an issue. The question is, how likely is it that we will have two different glaciers with the same name? I suppose if we only work on Svalbard, then it's probably quite low (but I don't know, you are the Svalbard glacier expert here😀).

If we need to handle this, each glacier has a unique IDENT (real data type). The part before the comma is unique for each glacier, the part behind the comma is unique for each part of the glacier. So here we see that Scheelebreen (13406) has three parts. This could be used to solve this issue.

<img width="153" alt="image" src="https://github.com/eliskasieglova/SvalbardSurges/assets/57137417/e8889a78-fddd-42be-85ca-ada3fb1609b7">

As long as we are only focusing on one glacier it is fine, but maybe for the future it would be better to choose the glacier not by NAME but by IDENT?

**Comment by erikmannerfelt** [comment]
Ah, thanks for the example! I would argue that these three outlines represent three different glaciers... There have been some updates in the naming of glaciers since these outlines were made. The second largest "Scheelebreen" is now called "Klubbebreen" (according to TopoSvalbard). The smallest one, however, is still missing a name. 

**But I would consider all disconnected outlines to be separate glaciers, for sure.**

I guess the importance of this debate will diminish when we scale up to work on regions or all of Svalbard. Then a misnomer here and there won't matter that much!

I guess a third workaround would be to use spatial indexing. Basically, "we want the glacier that intersects with this point in space". But again, it's only necessary when we work on individual glaciers.

**Comment by erikmannerfelt** [comment]
And yes, I agree. `IDENT` seems better than `NAME`.

**Comment by eliskasieglova** [comment]
Fixed in [c99d2c4](https://github.com/eliskasieglova/SvalbardSurges/commit/c99d2c474cb0d9637e89c72e304a5bdaffbb8d7a) by replacing NAME with IDENT.
